addons/arsenal/functions/fnc_scanConfig.sqf

```diff
@@ -282,11 +282,42 @@ uiNamespace setVariable [QGVAR(CBAdisposableLaunchers), compileFinal _launchers]
 uiNamespace setVariable [QGVAR(configItemsTools), compileFinal _toolList];
 
 // Compatibility: Override baseWeapon for RHS optics
-// No good way to do this via script for other attachments, needs manual compat
+// No good way to do this via script for other RHS attachments, needs manual compat
 private _baseWeaponCache = uiNamespace getVariable QGVAR(baseWeaponNameCache);
 {
     private _baseAttachment = configName (_cfgWeapons >> getText (_x >> "rhs_optic_base"));
     if (_baseAttachment != "") then {
-        _baseWeaponCache set [toLower configName _x, _baseAttachment];
+        _baseWeaponCache set [toLowerANSI configName _x, _baseAttachment];
     };
 } forEach ("getText (_x >> 'rhs_optic_base') != ''" configClasses _cfgWeapons);
+
+// Compatibility: Override baseWeapon for CBA Scripted Optics
+// Adapted from https://github.com/Theseus-Aegis/Mods/blob/master/addons/armory/functions/fnc_getBaseVariant.sqf
+private _isScriptedOptic = toString {
+    isClass (_x >> "CBA_ScriptedOptic") ||
+    {(getText (_x >> "weaponInfoType")) regexMatch "CBA_scriptedOptic.*?"}
```

Comment:
Why the `?` in the for the regex?

Reply:
Lazy matching

Reply:
What about adding the `/i` flag to make it case insensitive?

Reply:
That's where I got my info from - and I missed the big info box... xD